src/activityRepository.js

```diff
@@ -0,0 +1,46 @@
+class UserActivity {
+    constructor(activityData, userData) {
+        this.activityData = activityData
+        this.userData = userData
+    }
+
+    getUserActivityById = (id) => this.activityData.filter(data => data.userID === id)
+    getUserInfoById = (id) => this.userData.filter(data => data.id === id)
+
+    dailyMilesWalked(id, date) {
+        let userActivityData = this.getUserActivityById(id)
+        let userInfoData = this.getUserInfoById(id)
+        if(userActivityData[0].date === date) {
```

Comment:
So we need to pass a test that has a date through the parameter that does not strictly equal userActivityData[0].date just to show that it wont run without them being equal

Reply:
Yes exactly! I think the test would be checking that it returns `undefined` since it will never get to the return statement? You can see the `expect` test syntax here: [https://www.chaijs.com/api/bdd/#method_undefined](https://www.chaijs.com/api/bdd/#method_undefined)